### DISCUSSION TITLE
Can we use  a shared memory for each fuse damon thread to communicate with apps (i.e., the requester) for metadata (request data except io data) ?

### DISCUSSION THREAD

**Discussion by yangyun50** [libfuse/libfuse#967] [Q&A]
That is, I have seen the kernel patches [(fuse: fuse-over-io-uring)](https://lore.kernel.org/linux-mm/sxnfn6u4szyly7yu54pyhtg44qe3hlwjgok4xw3a5mr3r2vrwb@3lecpeavc2os/T/) proposed by @bsbernd  to improve performance of fuse. It is a great idea ! 
But what I am curious about is can we just use a shared memory (created by mmap) to communicate with apps? For example, for the FUSE_READ request, every fuse daemon thead can create a shared memory in the beginning using mmap, and fuse kernel can also access this memory area by the customized mmap. 
By doing so, when read a request from /dev/fuse, we can just put the request metadata (i.e., struct fuse_in_header and other request-related header (i.e., struct fuse_write_in for FUSE_WRITE)) in the shared memory and avoid the  extra copy in `splice_read` mode and extra iov_iter_get_pages operation, which degrades performance in `fuse_copy_args`.

**Comment by bsbernd** [comment]
It might give bit better performance, but /dev/fuse IO is general just too slow - I'm personally not too much interested to add these tiny optimizations when I already have patches that include that part and at the same time add in several other optimizations in one step. Besides that mmap adds initialization overhead and code - right now I'm figuring out what if there is a performance loss by not using mmap with these fuse-over-io-uring patches.
